### PR TITLE
Calculate metadata file location

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #8 
Added `MetadataKey` to calculate metadata file location.

### CODE REVIEW

**Review by genryxy — APPROVED** [review]
@olenagerasimova thanks, looks good to me

**Comment by olenagerasimova** [comment]
@g4s8 please, take a look/merge

File: src/main/java/com/artipie/cargo/meta/MetadataKey.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * Cargo metadata file name by name of the package. The rules are:<br/>
+ * - Packages with 1 character names are placed in a directory named 1.<br/>
+ * - Packages with 2 character names are placed in a directory named 2.
+ * - Packages with 3 character names are placed in the directory 3/{first-character}
+ * where {first-character} is the first character of the package name.<br/>
+ * - All other packages are stored in directories named {first-two}/{second-two} where the top
+ * directory is the first two characters of the package name, and the next subdirectory is the
+ * third and fourth characters of the package name. For example, cargo would be stored in a file
+ * named ca/rg/cargo. <br/>
```
Comment (g4s8) [inline src/main/java/com/artipie/cargo/meta/MetadataKey.java:18]:
```suggestion
 * Cargo metadata file name by name of the package. The rules are:
 * <ul>
 * <li>Packages with 1 character names are placed in a directory named {@code 1}.</li>
 * <li>Packages with 2 character names are placed in a directory named {@code 2}.</li>
 * <li>Packages with 3 character names are placed in the directory {@code 3/&#123;first-character&#125;},
 * where {@code &#123;first-character&#125;} is the first character of the package name.</li>
 * <li>All other packages are stored in directories named {@code &#123;first-two&#125;/&#123;second-two&#125;} where the top
 * directory is the first two characters of the package name, and the next subdirectory is the
 * third and fourth characters of the package name. For example, cargo would be stored in a file
 * named {@code ca/rg/cargo}.</li>
 * </ul>
```

File: src/main/java/com/artipie/cargo/meta/MetadataKey.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        final Key res;
+        if (this.pkg.length() == 1) {
+            res = new Key.From("1");
+        } else if (this.pkg.length() == 2) {
+            res = new Key.From("2");
+        } else if (this.pkg.length() == 3) {
+            res = new Key.From("3", this.pkg.substring(0, 1));
+        } else {
+            res = new Key.From(this.pkg.substring(0, 2), this.pkg.substring(2, 4));
+        }
```
Comment (g4s8) [inline src/main/java/com/artipie/cargo/meta/MetadataKey.java:53]: Isn't `switch`-`case` would be better here?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova please check some javadoc fixes and one question

**Comment by olenagerasimova** [comment]
@g4s8 thanks, corrected

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**